document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutor.java

```diff
@@ -82,20 +82,32 @@
   }
 
   private void logClauses(
-      Bson projection, Bson filterClause, Bson sortOrders, Pagination pagination) {
+      final Query query,
+      final Bson projection,
+      final Bson filterClause,
+      final Bson sortOrders,
+      final Pagination pagination) {
     log.debug(
-        "MongoDB find():\nCollection: {}\n Projections: {}\n Filter: {}\n Sorting: {}\n Pagination: {}",
+        "MongoDB find():\nQuery: {}\nCollection: {}\n Projections: {}\n Filter: {}\n Sorting: {}\n Pagination: {}",
+        query,
         collection.getNamespace(),
         projection,
         filterClause,
         sortOrders,
         pagination);
   }
 
-  private void logPipeline(List<BasicDBObject> pipeline) {
+  private void logPipeline(final List<BasicDBObject> pipeline) {
     log.debug(
         "MongoDB aggregation():\n Collection: {}\n Pipeline: {}",
         collection.getNamespace(),
         pipeline);
   }
+
+  private Query transformAndLog(Query query) {
+    log.debug("MongoDB query before transformation: {}", query);
```

Comment:
nit: as follow up, log only on exception or something. Otherwise, this will flood the number of logs or rate limit them.

Reply:
Sure